File: samples/client/petstore/java/src/main/java/org/openapijsonschematools/client/schemas/validation/BooleanSchemaValidator.java
package org.openapijsonschematools.client.schemas.validation;

import org.openapijsonschematools.client.configurations.SchemaConfiguration;
import org.openapijsonschematools.client.exceptions.InvalidTypeException;
import org.openapijsonschematools.client.exceptions.ValidationException;

import java.util.List;
import java.util.Set;

public interface BooleanSchemaValidator {
    boolean castToAllowedTypes(boolean arg, List<Object> pathToItem, Set<List<Object>> pathSet);
    boolean getNewInstance(boolean arg, List<Object> pathToItem, PathToSchemasMap pathToSchemas);
    boolean validate(boolean arg, SchemaConfiguration configuration) throws ValidationException, InvalidTypeException;
}

File: samples/client/petstore/kotlin/src/main/kotlin/org/openapijsonschematools/client/schemas/validation/BooleanSchemaValidator.kt
package org.openapijsonschematools.client.schemas.validation

import org.openapijsonschematools.client.configurations.SchemaConfiguration
import org.openapijsonschematools.client.exceptions.ValidationException

interface BooleanSchemaValidator<T> {
    @Throws(ValidationException::class)
    fun validate(arg: Boolean, configuration: SchemaConfiguration?): Boolean

    @Throws(ValidationException::class)
    fun validateAndBox(arg: Boolean, configuration: SchemaConfiguration?): T
}